### DISCUSSION TITLE
fixes for the DMs going skew-whiff?

### DISCUSSION THREAD

**Discussion by kitsunegari101** [AprilSylph/XKit-Rewritten#1155] [Suggestion Box]
Luckily I haven't received the "update" (yet), but Something has happened to the DMs; namely, the unnecessary white ring around the icons and an inability to close/minimize them unless I zoom out, at which point it's difficult at best to read anything. Not to mention the terrible new color contrast...eurgh. It's bad even if I try different color palettes.
![why is it this way CROP](https://github.com/AprilSylph/XKit-Rewritten/assets/14188767/736bdc56-c42b-4e98-8538-d1e576dd811d)

**Comment by marcustyphoon** [comment]
This isn't really a suggestion yet! If you momentarily pretend that the old UI didn't exist and list some things you would change about the new one, they could make good suggestions.

**Reply by kitsunegari101** [reply]
I did send some constructive feedback in after posting this! Fingers crossed it's more helpful than the scads of angry feedback staff is getting.

**Comment by zobothehob0** [comment]
PLEASE I need this fixed! So I'm in the minority that I don't mind the sidebar too much. If we could unsticky the search bar, it'd be an awesome change for me. I have vision issues and browse tumblr zoomed in to 120%. When zoomed in, the sidebar is smaller and less clunky and honestly, and awesome compliment to the dash. However... I have vision problems and the new DMs are a nightmare.

The standard text size online is 10pt, since this is too small for me, I have to zoom in. 12pt is my easiest view, hence me zooming into 120%. The new DM text size is 8pt. That's ridiculously small, about 1/4 smaller than I am comfortable reading. zoomed into 120%, they're 9.6. Still far too small for me to comfortably read. Zoomed into the largest zoom you can on firefox and have them on the screen (133%), they're still only 11.34pt. It is actually impossible to zoom the new DMs into 12pt font. This is not a lot, this is not a huge text size, this should be possible.

In addition, if you zoom in past 110%, the ability to minimize the chat disappears, zoomed out of view. That means that at 120%, the zoom I browse at, I cannot open a message and then close it. Since the messages now blog the reblog button even at 100%, making it now impossible to scroll and chat, not being able to close a chat window means you cannot use the dashboard. This means every time I open a message I have to zoom back out, minimize, and then zoom back in to resume use of my dash. This makes carrying on a conversation impossible.

I sent a message to tumblr support about the inaccessibility of this new layout and was frustrated and horrified with their response of "change is hard but you'll get used to it! :)" and I am desperate to find anything that can put the DMs back to what they were or fix the inaccessibility issues. Everyone I can see is only talking about scripts to change the sidebar, which I truly do not care about. This is one of the most frustrating weeks ever cause I truly cannot use tumblr and I feel like I've been kicked off the site for being too blind, and I'm not even low vision. I'm just "could maybe get glasses but hasn't because I can just zoom into my computer to make things 12pt and I'm fine." I can't imagine how actual low vision people feel right now. 

To be clear: I am suggesting a bit in accessibility or tweaks that allows reverting to old DMs or making the new DM box smaller (and increasing text size would be dope). In addition, there is no reason, imo, DMs have to be floating with that weird gap between them and the bottom of the page, rather than tabs like on fb and the old DMs. The issues that need fixing, however, are the zoom and the text size. Both of which were better in the old DMs.

**Reply by marcustyphoon** [reply]
The fact that the top of the DM window can go off the top of the screen is an obvious bug. I assume Staff will get on that; the simplest thing they could do is the equivalent of this (which you can put into [Stylus](https://github.com/openstyles/stylus#releases)) in the meantime:

```css
.j17Mp.be6E9 {
  max-height: calc(100vh - 90px);
}
```

The actual font size used in the DM window, both with the old layout and the new layout (the size was not changed), is 14 ~~point~~ px. That is, of course, 12.5% smaller than the 16 ~~point~~ px size used for text in posts on the dashboard. 14 ~~point~~ px (specifically, `0.875rem`)  being too small for you to comfortably read without zooming is good feedback, and I would hope that Staff would strongly consider increasing that font to the same size as that used in post content. If they don't, I would definitely consider an AccessKit checkbox for "increase the font size in DMs," though because this is still an experimental layout, I would push Staff on it first. (I'll send my own message.)

Note also that [Palettes for Tumblr](https://github.com/AprilSylph/Palettes-for-Tumblr#readme) provides a way to scale up the Tumblr font sizes **without** zooming and changing the size of everything. Obviously we want to push Staff to make the fonts big enough for everyone, though.

**Reply by zobothehob0** [reply]
Thank you SO much for the fixes wow! I know the font size is smaller, because one of my accounts doesn't have the update and I've compared them. It's also smaller on mobile (one of my phones was (accidentally) set to auto update and the other wasn't). I've also spoken to others who had an issue with how it got smaller. Like, it did get smaller. We're both wrong about the font size though. When I inspect the dash, the text size is 16px, which is 12pt. This is 12pt Arial next to the dash at 100% 
![image](https://github.com/AprilSylph/XKit-Rewritten/assets/140360487/f302354c-4c7d-4196-a04f-af94d9e29e80)

This is 11pt which I think looks like a match. But it should be noted, tumblr uses helvetica, which I do not have, which probably accounts for the discrepancy. 
![image](https://github.com/AprilSylph/XKit-Rewritten/assets/140360487/6349e51b-3908-4a2a-8d85-ccd9db64a315)

**Reply by marcustyphoon** [reply]
Ah, gotcha, px vs pt! My bad on the terminology there; I hadn't realized there was a difference.

And—that's weird; I currently have accounts side-by-side with and without the new messaging interface, and they definitely have the exact same font size... **but**, now that I think of it, both of those are accounts without the experimental dash layout. Maybe the combination of the experimental dash layout and the experimental messaging interface results in the CSS cascade being different and results in a different font size!

(If anyone sees this, has an account with both active, and is familiar enough with dev tools to know: I'd be curious if your message text is size `.875rem` due to the rule on `.hpABw`/"conversation" or if it's something else!)
